src/core/meshcop/meshcop_tlvs.cpp

```diff
@@ -211,6 +211,28 @@ void ChannelTlv::SetChannel(uint16_t aChannel)
     mChannel = HostSwap16(aChannel);
 }
 
+Error StateTlv::StateTlvToError(StateTlv::State aState)
```

Comment:
How about instead of converting to `Error we add a helper `StateToString()` here to convert a given `StateTlv`::State` to a string directly?
We can then use it to log from `HandleMgmtCommissionerSetResponse()`?